core/domain/classifier_services.py

```diff
@@ -182,15 +209,33 @@ def handle_trainable_states(
         state_training_jobs_mapping.validate()
         state_training_jobs_mappings.append(state_training_jobs_mapping)
 
-    classifier_models.StateTrainingJobsMappingModel.create_multi(
-        state_training_jobs_mappings)
+    mapping_models = []
+    for state_training_job_mapping in state_training_jobs_mappings:
+        instance_id = (
+            classifier_models.StateTrainingJobsMappingModel.get_entity_id(
+                state_training_job_mapping.exp_id,
+                state_training_job_mapping.exp_version,
+                state_training_job_mapping.state_name
+            )
+        )
+        mapping_instance = classifier_models.StateTrainingJobsMappingModel(
+            id=instance_id, exp_id=state_training_job_mapping.exp_id,
+            exp_version=state_training_job_mapping.exp_version,
+            state_name=state_training_job_mapping.state_name,
+            algorithm_ids_to_job_ids=(
+                state_training_job_mapping.algorithm_ids_to_job_ids
+            ))
+
+        mapping_models.append(mapping_instance)
+    models_to_put.extend(mapping_models)
+    return models_to_put
 
 
 def handle_non_retrainable_states(
     exploration: exp_domain.Exploration,
     state_names: List[str],
     exp_versions_diff: exp_domain.ExplorationVersionsDiff
-) -> List[str]:
+) -> tuple(List[str], List[classifier_models.StateTrainingJobsMappingModel]):
```

Comment:
Same note as above, and the docstring doesn't seem to be updated.

Reply:
The name of the function isn't changed... ("handle" is too vague anyway but we should be extra careful for stuff dealing with storage models)